lib/OpenQA/WebAPI/Plugin/Helpers.pm

```diff
@@ -396,6 +397,8 @@ sub _compose_job_overview_search_args ($c) {
     $v->optional('groupid')->num(0, undef);
     $v->optional('modules', 'comma_separated');
     $v->optional('limit', 'not_empty')->num(0, undef);
+    $v->optional('group_glob');
+    $v->optional('not_group_glob');
```

Comment:
I would leave out the "_glob" suffix.

Reply:
That would also have been my preference originally too, but we've already set a precedent here with `module_re` for regex. Feels kinda wrong now not to have a special suffix for globs too.

Reply:
Perl regexes are very powerful and compiling arbitrary strings from the internet into them has a lot of room for exploitation. For example i could probably use lookahead/behind to create a DoS attack even with small input strings. 🤔 Using globs would be a lot safer and easier to understand by users.

Reply:
Yes, unfortunately Perl has no good way of dealing with regexes from users. Unfortunately we already do this. I once wrote the `regex_match` helper so we at least don't get warnings in our server logs but a DoS attach is likely possible. So I guess using globbing is better here, indeed.

(We can actually always make use of native code and e.g. call pcre2. However, that's more work.)

Reply:
For more information see also https://owasp.org/www-community/attacks/Regular_expression_Denial_of_Service_-_ReDoS.